test/unit-test/dawn/SIR/TestSIRSerializer.cpp

```diff
@@ -1,54 +1,133 @@
-//                      / _` |/ _` \ \ /\ / / '_  | 
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
 //                     | (_| | (_| |\ V  V /| | | |
 //                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
 //
 //
-//  This file is distributed under the MIT License (MIT). 
+//  This file is distributed under the MIT License (MIT).
 //  See LICENSE.txt for details.
 //
 //===------------------------------------------------------------------------------------------===//
 
 #include "dawn/SIR/SIR.h"
 #include "dawn/SIR/SIRSerializer.h"
-#include "dawn/Unittest/ASTSimplifier.h"
-#include "dawn/Unittest/PrintAllExpressionTypes.h"
 #include <gtest/gtest.h>
 
 using namespace dawn;
-using namespace astgen;
 
 namespace {
 
-TEST(SIRSerializer, Serialize) {
-  SIR sir;
-  sir.Filename = "test";
-  
-  auto stencil = std::make_shared<sir::Stencil>();
-  stencil->Name = "foo";
-  stencil->Fields.emplace_back(std::make_shared<sir::Field>("foo", SourceLocation(10, 15)));
-  
-  auto stencilBodyAst = std::make_shared<AST>();
-  stencilBodyAst->setRoot(block(binop(lit("5"), "+", var("a")), assign(var("a"), var("b"))));
-  
-  auto vr = std::make_shared<sir::VerticalRegion>(
-      stencilBodyAst, std::make_shared<sir::Interval>(sir::Interval::Start, 12, 1, -3),
-      sir::VerticalRegion::LK_Forward);
-
-  auto stencilDescAst = std::make_shared<AST>();
-  stencilDescAst->setRoot(block(verticalRegion(vr)));
-  
-  stencil->StencilDescAst = stencilDescAst;
-  sir.Stencils.emplace_back(stencil);
-  
-  std::cout << SIRSerializer::serializeToString(&sir) << std::endl;
-}
-
-TEST(SIRSerializer, Deserialize) {
+#define SIR_EXCPECT_EQ(sir1, sir2)                                                                 \
+  do {                                                                                             \
+    auto comp = sir1->comparison(*sir2);                                                           \
+    EXPECT_TRUE(bool(comp)) << comp.why();                                                         \
+  } while(0);
 
+class SIRSerializerTest : public ::testing::Test {
+protected:
+  virtual void SetUp() override { sirRef = std::make_shared<SIR>(); }
+  virtual void TearDown() override { sirRef.reset(); }
+
+  std::shared_ptr<SIR> serializeAndDeserializeRef() {
+    return SIRSerializer::deserializeFromString(SIRSerializer::serializeToString(sirRef.get()));
+  }
+
+  std::shared_ptr<SIR> sirRef;
+};
+
+class SIRSerializerStencilTest : public SIRSerializerTest {
+  virtual void SetUp() override {
+    SIRSerializerTest::SetUp();
+
+    sirRef->Stencils.emplace_back(std::make_shared<sir::Stencil>());
+  }
+};
+
+TEST_F(SIRSerializerStencilTest, Name) {
+  sirRef->Stencils[0]->Name = "foo";
+  SIR_EXCPECT_EQ(sirRef, serializeAndDeserializeRef());
+}
+
+TEST_F(SIRSerializerStencilTest, SourceLocation) {
```

Comment:
The SIR ignores location in comparison, right? Therefore, this test seems to not be meaningful

Reply:
Indeed, fixed to do the check manually

Reply:
should we add a line of expected, gotten here?

Reply:
It don't think this is necessary as GTest provides such a line for us.

Reply:
but the line is 
`sirRef->Stencils[0]->Loc == serializeAndDeserializeRef()->Stencils[0]->Loc, expected true, got false`
which is not particualrly helpful. I would suggest adding the `<<` output method provided by GTest

Reply:
Well what more information do you want? This tells you precisely that the location do not match ;)